hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/OperationSubsystemStart.java

```diff
@@ -31,28 +33,43 @@
 
     @Override
     public void execute(OperationContext opContext, ModelNode model) throws OperationFailedException {
+
+        final AtomicReference<Thread> newThread = new AtomicReference<>();
+
         try {
             ServiceName name = SubsystemExtension.SERVICE_NAME;
             ServiceRegistry serviceRegistry = opContext.getServiceRegistry(true);
             MonitorService service = (MonitorService) serviceRegistry.getRequiredService(name).getValue();
 
             boolean restart = model.get("restart").asBoolean(false);
-            if (restart) {
-                LOGGER.debug("Asked to restart the Hawkular Monitor service. Will stop it, then restart it now.");
-                service.stopMonitorService();
-            } else {
-                LOGGER.debug("Asked to start the Hawkular Monitor service");
-            }
-            if (!service.isMonitorServiceStarted()) {
-                Thread newThread = new Thread(new Runnable() {
-                    @Override
-                    public void run() {
+            long delay = model.get("delay").asLong(0L);
+
+            newThread.set(new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    if (delay > 0) {
+                        try {
+                            Thread.sleep(delay);
+                        } catch (Exception e) {
+                            newThread.get().interrupt();
+                        }
+                    }
+
+                    if (restart && service.isMonitorServiceStarted()) {
+                        LOGGER.warn("Hawkular Monitor Service restart requested. Will stop and restart it now.");
+                        service.stopMonitorService();
+                    } else {
+                        LOGGER.warn("Asked to start the Hawkular Monitor service");
+                    }
+
+                    if (!service.isMonitorServiceStarted()) {
                         service.startMonitorService();
                     }
-                }, "Hawkular WildFly Agent Operation Start Thread");
-                newThread.setDaemon(true);
-                newThread.start();
-            }
+                }
+            }, "Hawkular WildFly Agent Operation Start Thread"));
+            newThread.get().setDaemon(true);
+            newThread.get().start();
+
         } catch (ServiceNotFoundException snfe) {
             throw new OperationFailedException("Cannot restart Hawkular Monitor service - it is disabled", snfe);
         } catch (Exception e) {

```

Comment:
you put the newThread interrupt in the wrong spot. You want a catch (InterruptedException ie) here just above the catch (Exception e). In that new catch clause is where you call newThread.get().interrupt(). That interrupt will (potentially) interrupt the sleep and when interrupting the sleep, you just simply "return". You want to tell the inner thread to stop doing what its doing - and you do taht by catching the interrupt and just returning.